### ISSUE TITLE
Class constructor Temporal cannot be invoked without 'new'

### ISSUE THREAD

**Issue by buschtoens** [js-joda/js-joda#215]
Using `.call(...)` on an ES2015 class directly is not supported.

```
Uncaught TypeError: Class constructor Temporal cannot be invoked without 'new'
```

https://github.com/js-joda/js-joda/blob/e8a7447a6ed0333af0d8de6be2197ca7695f7b06/src/LocalTime.js#L1390-L1398

```
Uncaught TypeError: Class constructor TemporalAmount cannot be invoked without 'new'
```

https://github.com/js-joda/js-joda/blob/e7967e6a82d0faf2a5b331005974dc5bb5f20fce/src/Period.js#L928-L935

This problem does not arise in the dist build, since the ES2015 classes are transpiled to ES5 compatible `function` / `prototype` code.

However, I am using the original source code in [`ember-joda`](https://github.com/buschtoens/ember-joda) to apply some minor compatibility changes before transpiling it. Depending on the targets, some transforms, like the [`transform-es2015-classes`](https://babeljs.io/docs/plugins/transform-es2015-classes/), are disabled.

This causes the `TypeError`s above.

My current workaround is to always run `transform-es2015-classes`, however, it'd be great, if we could make sure that the actual `js-joda` source code is valid without transpilation.

**Comment by pithu** [comment]
Indeed, quite ugly code pointers. Does the commit above solve your issue ?

**Comment by pithu** [comment]
fyi: i just released js-joda@1.6.3 without any `call on constructors`

**Comment by buschtoens** [comment]
Works perfectly. Thanks! :grinning:
